packages/test/test-end-to-end-tests/src/test/gc/gcLoadingFromOlderSummaries.spec.ts

```diff
@@ -0,0 +1,173 @@
+    let mainContainer: IContainer;
+    let containerRuntime: IContainerRuntime;
+    let dataStoreA: ITestDataObject;
+
+    /**
+     * Submits a summary and returns the reference state for all the nodes in the container.
+     * If a node is referenced, its value is true. If it's unreferenced, its value is false.
+     * @returns a map of nodePath to its unreferenced timestamp.
+     */
+    async function getReferenceState(summaryTree: ISummaryTree) {
+        const gcState = getGCStateFromSummary(summaryTree);
+        assert(gcState !== undefined, "GC tree is not available in the summary");
+
+        const nodeReferenceStates: Map<string, boolean> = new Map();
+        for (const [nodePath, nodeData] of Object.entries(gcState.gcNodes)) {
+            nodeReferenceStates.set(nodePath.slice(1), nodeData.unreferencedTimestampMs !== undefined ? false : true);
+        }
+        return nodeReferenceStates;
+    }
+
+    /*
+     * Utility function that returns the sequence number of a summary from the summary metadata.
+     */
+    function getSummarySequenceNumber(summaryTree: ISummaryTree) {
+        const metadataBlob = summaryTree.tree[".metadata"];
+        assert(metadataBlob.type === SummaryType.Blob, "Container runtime metadata is not a blob");
+        const metadata = JSON.parse(metadataBlob.content as string) as Record<string, unknown>;
+        return (metadata.message as ISequencedDocumentMessage).sequenceNumber;
+    };
+
+    /**
+     * Function that asserts the given value is not true. Used in these tests to demonstrate that because of
+     * a bug we are not getting the expected results. Once the bug is fixed, these asserts should start working
+     * as expected.
+     */
+    function assertNotTrue(value: boolean, message: string) {
+        assert(!value, message);
+    }
+
+    beforeEach(async function() {
+        provider = getTestObjectProvider({ syncSummarizer: true });
+        // These tests validate the GC state in summary generated by the container runtime. They do not care
+        // about the snapshot that is downloaded from the server. So, it doesn't need to run against real services.
+        if (provider.driver.type !== "local") {
+            this.skip();
+        }
+
+        mainContainer = await provider.makeTestContainer(defaultGCConfig);
+        dataStoreA = await requestFluidObject<ITestDataObject>(mainContainer, "default");
+        containerRuntime = dataStoreA._context.containerRuntime as IContainerRuntime;
+        await waitForContainerConnection(mainContainer);
+    });
+
+    it("updates referenced nodes correctly when loading from an older summary", async () => {
+        const summarizer1 = await createSummarizer(provider, mainContainer);
+
+        // Create a data store and mark it unreferenced to begin with.
+        const dataStoreB = await requestFluidObject<ITestDataObject>(
+            await containerRuntime.createDataStore(TestDataObjectType), "");
```

Comment:
Can you just get the handle of the datastore here and call get?

Reply:
Did the change go in where createDataStore returns a handle? If so, yes we can do that.

Reply:
I think so

Reply:
Changed it.